### PR TITLE
MGMT-6663: Update progress of operators only when there's a new progress

### PR DESCRIPTION
**Pull request by YuviGold**

### CODE REVIEW

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 1]
```diff
@@ -772,13 +772,15 @@ func (c controller) waitForOLMOperators() bool {
 		}
 
 		operatorStatus := utils.CsvStatusToOperatorStatus(string(csv.Status.Phase))
-		err = c.ic.UpdateClusterOperator(context.TODO(), c.ClusterID, operator.Name, operatorStatus, csv.Status.Message)
-		if err != nil {
-			c.log.WithError(err).Warnf("Failed to update olm %s status", operator.Name)
-			continue
-		}
+		if operator.Status != operatorStatus || (operator.StatusInfo != csv.Status.Message && csv.Status.Message != "") {
```
Comment (gamli75) [inline src/assisted_installer_controller/assisted_installer_controller.go:775]: why we need the "&& csv.Status.Message != " in the condition? I think that we will skip updates - I guess that the default for csv.Status.Message is an empty string

Comment (YuviGold) [reply src/assisted_installer_controller/assisted_installer_controller.go:775]: In case the message is empty there is nothing to update

**Comment by YuviGold** [comment]
/cc @tsorya  @eranco74 
Please take a look.

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 2]
```diff
@@ -799,13 +799,15 @@ func (c controller) waitForOLMOperators() bool {
 			continue
 		}
 
-		err = c.ic.UpdateClusterOperator(context.TODO(), c.ClusterID, operator.Name, operatorStatus, csv.Status.Message)
-		if err != nil {
-			c.log.WithError(err).Warnf("Failed to update olm %s status", operator.Name)
-			continue
-		}
+		if operator.Status != operatorStatus || (operator.StatusInfo != csv.Status.Message && csv.Status.Message != "") {
+			c.log.Infof("CSV %s is in status %s, message %s.", operator.Name, csv.Status.Phase, csv.Status.Message)
```
Comment (eranco74) [inline src/assisted_installer_controller/assisted_installer_controller.go:803]: Consider logging the change
```suggestion
			c.log.Infof("CSV %s updated, status: %s -> %s, message: %s -> %s.", operator.Name, operator.Status, csv.Status.Phase, operator.StatusInfo, csv.Status.Message)
```

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 3]
```diff
@@ -818,21 +820,34 @@ func (c controller) isOperatorAvailableInCluster(operatorName string) bool {
...
+	operatorStatus, operatorMessage := utils.ClusterOperatorConditionsToMonitoredOperatorStatus(co.Status.Conditions)
+	if operatorStatusInService.Status != operatorStatus || (operatorStatusInService.StatusInfo != operatorMessage && operatorMessage != "") {
+		c.log.Infof("Operator %s status: %s message: %s", operatorName, operatorStatus, operatorMessage)
 
-	c.log.Infof("%s operator is available in cluster", operatorName)
+		err = c.ic.UpdateClusterOperator(context.TODO(), c.ClusterID, operatorName, operatorStatus, operatorMessage)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to update %s operator status %s with message %s", operatorName, operatorStatus, operatorMessage)
+			return false
+		}
```
Comment (eranco74) [inline src/assisted_installer_controller/assisted_installer_controller.go:837]: Consider extracting to a separate method and call it from waitForOLMOperators and here

File: src/assisted_installer_controller/assisted_installer_controller.go (hunk 2 above)
Comment (YuviGold) [reply src/assisted_installer_controller/assisted_installer_controller.go:803]: Good idea!

File: src/assisted_installer_controller/assisted_installer_controller.go (hunk 3 above)
Comment (YuviGold) [reply src/assisted_installer_controller/assisted_installer_controller.go:837]: I opened a ticket for merging those
https://issues.redhat.com/browse/MGMT-7149

**Comment by YuviGold** [comment]
```
time="2021-07-05T08:13:35Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:13:35Z" level=info msg="CSV lso updated, status:  -> available, message:  -> install strategy completed with no errors."
time="2021-07-05T08:13:35Z" level=warning msg="Operator ocs has failed state retry(1/10) the check."
time="2021-07-05T08:14:05Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:14:08Z" level=info msg="CSV ocs updated, status:  -> progressing, message:  -> installing: waiting for deployment ocs-operator to become ready: deployment \"ocs-operator\" not available: Deployment does not have minimum availability.."
time="2021-07-05T08:14:35Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:15:05Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:15:35Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:15:49Z" level=info msg="Start uploading controller logs (intermediate snapshot)"
time="2021-07-05T08:15:49Z" level=info msg="Uploading logs for assisted-installer-controller-c25mc in assisted-installer"
time="2021-07-05T08:15:50Z" level=info msg="Done uploading controller logs to the service"
time="2021-07-05T08:16:05Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:16:35Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:17:05Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:17:35Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:18:05Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:18:21Z" level=info msg="CSV ocs updated, status: progressing -> available, message: installing: waiting for deployment ocs-operator to become ready: deployment \"ocs-operator\" not available: Deployment does not have minimum availability. -> install strategy completed with no errors."
time="2021-07-05T08:18:35Z" level=info msg="Checking OLM operators"
time="2021-07-05T08:18:49Z" level=info msg="Start complete installation step, with params success:true, error info "
```

**Pull request merged**